qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -89,9 +96,20 @@ class HighLevelSynthesis(TransformationPass):
     ``default`` methods for all other high-level objects, including ``op_a``-objects.
     """
 
-    def __init__(self, hls_config=None):
+    def __init__(self, coupling_map: CouplingMap = None, hls_config=None):
+        """
+        HighLevelSynthesis initializer.
+
+        Args:
+            coupling_map (CouplingMap): the coupling map of the backend
+                in case synthesis is done on a physical circuit.
```

Comment:
Can we also add a `Target` argument here, similar to what I do in #9263 ? Basically I'm trying to unify our internal transpiler model around the target. My plan for 0.24.0 is to hopefully have the preset pass manager only use a target internally.

Reply:
@mtreinish, so should right now `HighLevelSynthesis` accept both `coupling_map` and `target`, or just `target`? I guess I am asking whether `generate_translation_passmanager` (or code further upstream) updates `target` with `coupling_map` information if only the latter is specified.

Reply:
Right now it needs to accept both, there is still a code path in `transpile` where a target will not be generated (I was hoping to change this for 0.24, so it was only target, but it won't make it). You can do it with a second argument or what we did in #9263 is had a single argument take either a coupling map or a target.